src/odemis/gui/cont/project.py

```diff
@@ -0,0 +1,431 @@
+        # technolution driver
+        roa_project = [p for p in self._tab_data.projects.value if self.model in p.roas.value][0]
+        all_project_roas = [roa.name.value for roa in roa_project.roas.value]
+        if txt == "":
+            txt = current_name
+            self.panel.txt_ctrl.SetValue(txt)
+        if txt != current_name:
+            txt = make_unique_name(txt, all_project_roas)
+            logging.debug("Renaming ROA from %s to %s.", self.model.name.value, txt)
+            self.model.name.value = txt
+            self.panel.txt_ctrl.SetValue(txt)
+
+        evt.Skip()
+
+
+class FastEMROCController(object):
+    """
+    Controller for a single region of calibration.
+    """
+
+    def __init__(self, number, tab_data, view_ctrl):
+        """
+        number (int): number of the calibration region
+        tab_data (FastEMAcquisitionGUIData): tab data model
+        view_ctrl (FastEMAcquisitionViewport): viewport controller
+        """
+        self._view_ctrl = view_ctrl
+        self._tab_data = tab_data
+
+        # By default, the calibration region is in the center of the scintillator. The size is given by
+        # the ebeam resolution.
+        pos = tab_data.main.scintillator_positions[number]
+        sz = (tab_data.main.ebeam.resolution.value[0] * tab_data.main.ebeam.pixelSize.value[0],
+              tab_data.main.ebeam.resolution.value[1] * tab_data.main.ebeam.pixelSize.value[1])
+
+        l = pos[0] - 0.5 * sz[0]
+        t = pos[1] + 0.5 * sz[1]
+        r = pos[0] + 0.5 * sz[0]
+        b = pos[1] - 0.5 * sz[1]
+
+        # Get ROC model (exists already in tab data) and change coordinates
+        self.model = tab_data.calibration_regions.value[number]
+        self.model.coordinates.value = (l, t, r, b)
+        self.model.coordinates.subscribe(self._on_coordinates)
+
+        # Add overlay
+        self.overlay = view_ctrl.viewports[0].canvas.add_calibration_overlay(self.model.coordinates, number)
+
+    def fit_view_to_bbox(self):
+        """
+        Zoom in to calibration region. Calibration region is in the center and takes up 1/100 of viewport area.
+        """
+        logging.debug("Zooming in to calibration region %s.", self.model.name.value)
+        cnvs = self._view_ctrl.viewports[0].canvas
+        l, t, r, b = self.model.coordinates.value
+        sz = (self._tab_data.main.ebeam.resolution.value[0] * self._tab_data.main.ebeam.pixelSize.value[0],
+              self._tab_data.main.ebeam.resolution.value[1] * self._tab_data.main.ebeam.pixelSize.value[1])
+        cnvs.fit_to_bbox([l - 5 * sz[0], t - 5 * sz[1], r + 5 * sz[0], b + 5 * sz[1]])
+
+    def _on_coordinates(self, coords):
+        # Purely for logging
+        logging.debug("ROC '%s' coordinates changed to %s.", self.model.name.value, coords)
+
+
+class FastEMCalibrationRegionsController(object):
+    """
+    Listens to the calibration buttons and creates the FastEMROCControllers accordingly.
```

Comment:
I know this is just copied from when it was still in streams.py, but the term "calibration buttons" is a bit vague. Maybe there it's also better to call them "calibration region buttons", especially now that we also have the Calibrate buttons.

Reply:
Would it be ok to do this in the next pr? I have might need to modify more and/or can make the doc strings etc also a bit more explanatory?

Reply:
no problem!